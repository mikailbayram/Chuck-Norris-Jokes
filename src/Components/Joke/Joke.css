.joke:first-child{
    margin-top:100px;
    width: 70%;
    margin-left: 15%;
    opacity: 1;
    -webkit-animation: slide-up-fade-in-first ease 1s;
    animation: slide-up-fade-in-first ease 1s;
}
@keyframes slide-up-fade-in-first{
    0% {
        opacity:0;
    }
    100% {
        opacity:1;
    }
}
.joke{
    width: 50%;
    margin-left: 25%;
    margin-bottom: 50px;
    opacity: 0.5;
    -webkit-animation: slide-up-fade-in ease 1s;
    animation: slide-up-fade-in ease 1s;
    cursor: pointer;
}
.joke:not(:first-child):hover{
    transform: scaleX(1.2);
    opacity: 1;
}
@keyframes slide-up-fade-in{
    0% {
        opacity:0;
        transform:  translate(0px,40px)  ;
    }
    100% {
        opacity:0.5;
        transform:  translate(0px,0px)  ;
    }
}
@-webkit-keyframes slide-up-fade-in{
    0% {
        opacity:0;
        transform:  translate(0px,40px)  ;
    }
    100% {
        opacity:0.5;
        transform:  translate(0px,0px)  ;
    }
}
.joke-text{
    font-size: 1.5em;
    width: 80%;
    margin-top: 20px;
    color: gray;
    font-style: italic;
}
.quote-icon{
    width: 5%;
    font-size: 2em;
    margin-left: 5%;
}
.joke-text-wrapper{
    display: flex;
    justify-content: flex-start;
}
.posted-at{
    float: right;
    top:10px;
    right: 10px;
    position: absolute;
    color:gray; 
    font: 'Roboto';
}
