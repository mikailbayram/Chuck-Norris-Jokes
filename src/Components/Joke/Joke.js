import React, { Component } from 'react';
import {Card, CardHeader, CardText} from 'material-ui/Card';
import Modal from "../Modal/Modal"
import MdFormatQuote from "react-icons/lib/md/format-quote";
import "./Joke.css";

class Joke extends Component {
  constructor(props){
      super(props);
      this.state = {
          open:false,
      }
      this.close=this.close.bind(this);
  }
  close(){
      this.setState({open:false})
  }
  render() {
    return (
            <Card className="joke" onClick={()=>this.setState({open:true})}>
                <CardHeader
                title="Juck Norris"
                subtitle={!this.props.isHome?this.props.joke.get("category").get(0):"Random"}
                avatar={this.props.icon}
                children={<div className="posted-at">
                    {this.props.joke.get("posted")}
                    <br/>
                    {!this.props.isHome&&<span>Times Repeated: {this.props.joke.get("times_repeated")}</span>}
                </div>}
                />
                <CardText className="joke-text-wrapper">
                    <MdFormatQuote className="quote-icon"/>
                    {this.props.text?<div className="joke-text" dangerouslySetInnerHTML={{__html: this.props.text}}/>
                    :<div className="joke-text">{this.props.joke.get("value")}</div>}
                    {this.state.open&&<Modal close={this.close}
                    icon={this.props.icon} 
                    open={this.state.open} 
                    joke={this.props.joke}
                    path={this.props.path}/>}
                </CardText>
            </Card>
       
    );
  }
}

export default Joke;
