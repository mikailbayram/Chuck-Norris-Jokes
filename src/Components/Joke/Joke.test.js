import React from 'react';
import ReactDOM from 'react-dom';
import Joke from './Joke';
import {configure, shallow} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';
import {Map,List} from 'immutable';
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';

configure({ adapter: new Adapter() });

describe('<Joke />', () => {
   
    let joke;
    let wrapper;
    beforeAll(()=>{
        joke = Map({
            category:List(["Dev"])
        })
        wrapper = shallow(<Joke joke={joke} />);
    })
    it('renders without crashing',()=>{
        expect(wrapper.state().open).toEqual(false);
    })
    it('runs the close method',()=>{
        wrapper.instance().close();
        expect(wrapper.state().open).toEqual(false);
    })
    it('changes state on click', () => {
        const card = wrapper.find("Card").at(0);
        card.simulate("click");
        expect(wrapper.state().open).toEqual(true);
    });
    it('renders branches', ()=>{
        wrapper = shallow(<Joke joke={joke} isHome={true} text={"some text"}/>);
        expect(wrapper.state().open).toEqual(false);
    })
    it('renders correctly', () => {
        const tree = renderer
          .create(
          <MuiThemeProvider>
              <Joke joke={joke} />
          </MuiThemeProvider>)
          .toJSON();
        expect(tree).toMatchSnapshot();
      });
});
