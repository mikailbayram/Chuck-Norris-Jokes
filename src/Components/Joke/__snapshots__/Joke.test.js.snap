// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Joke /> renders correctly 1`] = `
<div
  className="joke"
  onClick={[Function]}
  style={
    Object {
      "WebkitTapHighlightColor": "rgba(0,0,0,0)",
      "backgroundColor": "#ffffff",
      "borderRadius": 2,
      "boxShadow": "0 1px 6px rgba(0, 0, 0, 0.12),
             0 1px 4px rgba(0, 0, 0, 0.12)",
      "boxSizing": "border-box",
      "color": "rgba(0, 0, 0, 0.87)",
      "fontFamily": "Roboto, sans-serif",
      "muiPrepared": true,
      "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
      "zIndex": 1,
    }
  }
>
  <div
    style={
      Object {
        "paddingBottom": 8,
      }
    }
  >
    <div
      style={
        Object {
          "boxSizing": "border-box",
          "fontWeight": 500,
          "muiPrepared": true,
          "padding": 16,
          "position": "relative",
          "whiteSpace": "nowrap",
        }
      }
    >
      <div
        style={
          Object {
            "display": "inline-block",
            "muiPrepared": true,
            "paddingRight": "90px",
            "verticalAlign": "top",
            "whiteSpace": "normal",
          }
        }
      >
        <span
          style={
            Object {
              "color": "rgba(0, 0, 0, 0.87)",
              "display": "block",
              "fontSize": 15,
              "muiPrepared": true,
            }
          }
        >
          Juck Norris
        </span>
        <span
          style={
            Object {
              "color": "rgba(0, 0, 0, 0.54)",
              "display": "block",
              "fontSize": 14,
              "muiPrepared": true,
            }
          }
        >
          Dev
        </span>
      </div>
      <div
        className="posted-at"
      >
        <br />
        <span>
          Times Repeated: 
        </span>
      </div>
    </div>
    <div
      className="joke-text-wrapper"
      style={
        Object {
          "color": "rgba(0, 0, 0, 0.87)",
          "fontSize": 14,
          "muiPrepared": true,
          "padding": 16,
        }
      }
    >
      <svg
        className="quote-icon"
        fill="currentColor"
        height="1em"
        preserveAspectRatio="xMidYMid meet"
        style={
          Object {
            "color": undefined,
            "verticalAlign": "middle",
          }
        }
        viewBox="0 0 40 40"
        width="1em"
      >
        <g>
          <path
            d="m23.4 28.4l3.2-6.8h-5v-10h10v10l-3.2 6.8h-5z m-13.4 0l3.4-6.8h-5v-10h10v10l-3.4 6.8h-5z"
          />
        </g>
      </svg>
      <div
        className="joke-text"
      />
    </div>
  </div>
</div>
`;
